src/lib.rs

```diff
@@ -105,11 +127,141 @@ impl Fairing for RocketSentry {
                 if config.sentry_dsn.is_empty() {
                     info!("Sentry disabled.");
                 } else {
-                    self.init(&config.sentry_dsn);
+                    let traces_sample_rate = config.sentry_transaction_sample_rate.unwrap_or(0f32);
+                    self.init(&config.sentry_dsn, traces_sample_rate);
                 }
             }
             Err(err) => error!("Sentry not configured: {}", err),
         }
         Ok(rocket)
     }
+
+    async fn on_request(&self, request: &mut Request<'_>, _: &mut Data<'_>) {
+        let name = request_to_transaction_name(request);
+        let build_transaction = move || Self::start_transaction(&name);
+        let request_transaction = local_cache_once!(request, build_transaction);
+        request.local_cache(request_transaction);
+    }
+
+    async fn on_response<'r>(&self, request: &'r Request<'_>, response: &mut Response<'r>) {
+        // We take the transaction set in the on_request callback
+        let request_transaction = local_cache_once!(request, Self::invalid_transaction);
+        let ongoing_transaction: &Transaction = request.local_cache(request_transaction);
+        ongoing_transaction.set_status(map_status(response.status()));
+        set_transaction_request(ongoing_transaction, request);
+        ongoing_transaction.clone().finish();
+    }
+}
+
+fn set_transaction_request(transaction: &Transaction, request: &Request) {
+    transaction.set_request(protocol::Request {
+        url: None,
+        method: Some(String::from(request.method().as_str())),
+        data: None,
+        query_string: request_to_query_string(request),
+        cookies: None,
+        headers: Default::default(),
+        env: Default::default(),
+    });
+}
+
+fn request_to_transaction_name(request: &Request) -> String {
+    let method = request.method();
+    let path = request.uri().path();
+    format!("{method} {path}")
+}
+
+fn request_to_query_string(request: &Request) -> Option<String> {
+    let query_string = request.uri().query()?.as_str().to_string();
+    Some(query_string)
+}
+
+fn map_status(status: Status) -> protocol::SpanStatus {
+    match status.code {
+        100..=299 => protocol::SpanStatus::Ok,
+        300..=399 => protocol::SpanStatus::InvalidArgument,
```

Comment:
In practice most `3xx` statuses are redirects. `InvalidArgument` seems a bit misleading, are there better options?

I think weshould look at how other Sentry implementations map status codes.

Reply:
I agree `InvalidArgument` is a bit vague and is reused for `4xx`. Also SpanStatus explicitly states it is for 4xx: https://github.com/getsentry/sentry-rust/blob/master/sentry-types/src/protocol/v7.rs#L1846.
I used [actix](https://github.com/getsentry/sentry-rust/blob/master/sentry-actix/src/lib.rs#L321) as a global model, but `3xx` indeed map to [redirection](https://docs.rs/actix-web/latest/actix_web/http/struct.StatusCode.html#method.is_redirection) there, and on sentry this case is not handled and they default to `UnknownError`.

On Python they rather include it in the OK status and I think it's the best option: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/tracing.py#L408.

Reply:
The actix one seems to return `UnknownError` instead of `InvalidArgument`?

Anyway I'd opt to follow the Python implementation, let's add a link to it in code comment as well:

https://github.com/getsentry/sentry-python/blob/e0d7bb733b5db43531b1efae431669bfe9e63908/sentry_sdk/tracing.py#L408-L435

Reply:
Done.